samza-core/src/main/java/org/apache/samza/SamzaContainerStatus.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza;
+
+
+/**
+ * <pre>
+ *                                                                   runloop completed [OR]
+ *                  container.run()           runloop.run            container.shutdown()
+ *    NOT_STARTED -----------------> STARTING ------------> RUNNING -----------------------> STOPPED
+ *                                       |      Error in runloop |
+ *                                       |      [OR] Error when  |
+ *                           Error when  |      stopping         |
+ *                   starting components |      components       |
+ *                                       V                       |
+ *                                    FAILED <-------------------|
+ * </pre>
+ */
+
+/**
+ * Indicates the current status of a {@link org.apache.samza.container.SamzaContainer}
+ */
+public enum  SamzaContainerStatus {
+  /**
+   * Indicates that the container has not been started
+   */
+  NOT_STARTED,
```

Comment:
ApplicationStatus should rely on the status of the StreamProcessor itself, and not on its individual components. Having said that, we have not defined states for StreamProcessors yet. We should, ideally, define states for StreamProcessor and that should be inspected by the ApplicationRunner for status. 

I would prefer to keep SamzaContainer's initial status as NOT_STARTED because each time we rebalance, we are creating a "new" samza container instance, which is technically not started. Calling it NEW really doesn't make sense atm. 

Is there any state transition details for ApplicationStatus? What it means in the standalone scenarios?